### DISCUSSION TITLE
Overcoming -deskew changes in new version 7.0.11-2

### DISCUSSION THREAD

**Discussion by 00SS** [ImageMagick/ImageMagick#3347] [Help] [answered]
I'm trying to get as small as possible PNG files, as I have a few hundred large B/W images of lines of text that need to be processed into a PDF file.

Using: ImageMagick-7.0.11-2-Q16-x64-dll.exe (Windows 10)
```"C:\Program Files\ImageMagick-7.0.11-Q16\magick.exe" logo:  -monochrome -deskew 40% -depth 1 -strip -quality 99 logo_7011.png```
Result: logo_7011.png : 5,848 bytes

Using: ImageMagick-7.0.7-11-Q16-x64-dll.exe
```"C:\Program Files\ImageMagick-7.0.7-Q16\magick.exe" logo:  -monochrome -deskew 40% -depth 1  -strip -quality 99 logo_707.png```
Result: logo_707.png : 8,872 bytes

Although this particular PNG created in version 7.0.11-2 is smaller, it is not usable - as compared to the larger good PNG created by version 7.0.7-11. When deskewing text though, the newer version is giving (after reducing the colours to B/W) larger PNG files.

I believe the difference is happening as ```-deskew``` in the new version changes the image from B/W to grayscale (this didn't happen in the older version).

How could I improve the PNG when using ```-deskew``` in version 7.0.11-2 - and also in a way that would hopefully not break when newer versions of imagemagick are released.

**Comment by snibgo** [comment, marked as answer]
I'm not sure exactly what "-monochrome" does. In v7.0.8-64 it was not sensitive to dither settings, but in v7.0.10-45 it is, so "+dither -monochrome" may reproduce the behaviour you used to see.

Personally, I don't use "-monochrome". Instead, I would use "-colorspace Gray -threshold 50%".

**Reply by 00SS** [reply]
Thanks. ```-threshold 50%``` gives the result wanted, and is better than ```-depth 1```
